src/aggregator/diff-aggregator.ts

```diff
@@ -248,9 +248,30 @@ export class DiffAggregator implements Aggregator {
   private isEqualCustomizer(c: any, s: any): boolean | undefined {
     if (_.isString(c) && c === GeneratedValue.COLOR) {
       return true; // always accept stored value if collected value is generated
-    } else if (_.isArray(c) && _.isArray(s) && c.length && _.isString(c[0])) {
+    }
+
+    if (_.isArray(c) && _.isArray(s) && c.length && _.isString(c[0])) {
       return _.isEqual(c.sort(), s.sort());
     }
+
+    if (_.isPlainObject(c) && _.isPlainObject(s)) {
+      const filteredC = this.filterUndefinedDeep(c);
+      const filteredS = this.filterUndefinedDeep(s);
+      return _.isEqual(filteredC, filteredS);
```

Comment:
There can be a problem with the `_.isEqual` calls here and a few lines above in the if block with the isArray checks:
The isEqualCustomizer logic is not applied anymore recursively. For example here if c or s has an array property, they won't be sorted anymore.
Shouldn't you use
```
          !_.isEqualWith(c, s, (cc, ss) =>
            this.isEqualCustomizer(cc, ss),
          )
```
here and above?

Reply:
I reworked it a bit. Now the filtering is before the equal checking. The sorting issue is unrelated here, but valid, I won't fix it in this PR, but I'll create a ticket for it